Libraries/LibCore/EventLoop.h

```diff
@@ -92,7 +100,57 @@ class EventLoop {
         NonnullOwnPtr<Event> event;
     };
 
+    struct SignalHandlers {
+        AK_MAKE_NONCOPYABLE(SignalHandlers);
+        friend class EventLoop;
+        friend class HashMap<int, SignalHandlers>;
```

Comment:
Uhh, why is this friendship necessary?

Reply:
The friendship for HashMap is needed because because `HashMap::remove` requires access to the `SignalHandlers` destructor, which is `private` because the entire class is considered `private`:
```
././AK/HashMap.h:58:68: error: 'Core::EventLoop::SignalHandlers::SignalHandlers(Core::EventLoop::SignalHandlers&&)' is private within this context
   58 |     HashSetResult set(const K& key, V&& value) { return m_table.set({ key, move(value) }); }
```

And the friendship for `EventLoop` is needed for this:
```
../Libraries/LibCore/EventLoop.cpp:466:34: error: 'void Core::EventLoop::SignalHandlers::dispatch()' is private within this context
  466 |         handlers->value.dispatch();
```

Reply:
why is a struct private?
does `AK_MAKE_NONCOPYABLE` define a private ctor?

Reply:
oh it leaks the `private`-ness too...
that needs fixing, @awesomekling:
```c++
#define AK_MAKE_NONCOPYABLE(c) \
private:                       \
    c(const c&) = delete;      \
    c& operator=(const c&) = delete;
```

should probably be (for structs):
```c++
#define AK_MAKE_NONCOPYABLE(c) \
private:                       \
    c(const c&) = delete;      \
    c& operator=(const c&) = delete; \
public:
```

is there a way to detect if some type is defined as a struct or a class...?

Reply:
Ah that's a good catch, I didn't realize these macros mess with visibility ;-)

Reply:
So how should we resolve this? I'm thinking that it shouldn't be necessary to mock with the visibility in these macros, because either way these functions are being deleted so there's no way code that needs them could compile, regardless of visibility. Or am I missing something?

Reply:
What we normally do is just put a `public:` after them, e.g:

```cpp
class Foo {
    AK_MAKE_NONCOPYABLE(Foo);
public:
    ...
};
```